Sources/NIO/AsyncAwaitSupport.swift

```diff
@@ -0,0 +1,92 @@
+#if compiler(>=5.4) && $AsyncAwait
+extension EventLoopFuture {
```

Comment:
for @weissi : make promise constructor that takes async function (thanks @tachyonics)

Reply:
@tachyonics now done